packages/components/namespace-store/reducer.tsx

```diff
@@ -0,0 +1,72 @@
+import { AWS_REGIONS } from "../../constants";
+
+export type ProviderDataState = {
+  secretName: string;
+  secretKey: string;
+  accessKey: string;
+  region: string;
+  target: string;
+  endpoint: string;
+};
+
+export const initialState: ProviderDataState = {
+  secretName: '',
+  secretKey: '',
+  accessKey: '',
+  region: AWS_REGIONS[0],
+  target: '',
+  endpoint: '',
+};
+
+export type StoreAction =
+  | { type: 'setSecretName'; value: string }
+  | { type: 'setSecretKey'; value: string }
+  | { type: 'setAccessKey'; value: string }
+  | { type: 'setRegion'; value: string }
+  | { type: 'setTarget'; value: string }
+  | { type: 'setEndpoint'; value: string };
+
+export type BackingStoreProviderDataState = ProviderDataState & {
+  numVolumes: number;
+  gcpJSON: string;
+  volumeSize: string;
+  storageClass: string;
+};
+
+export type BackingStoreAction =
+  | StoreAction
+  | { type: 'setGcpJSON'; value: string }
+  | { type: 'setVolumes'; value: number }
+  | { type: 'setVolumeSize'; value: string }
+  | { type: 'setStorageClass'; value: string };
+
+export const providerDataReducer = (
+  state: BackingStoreProviderDataState,
+  action: BackingStoreAction,
```

Comment:
```suggestion
  state: ProviderDataState,
  action: StoreAction,
```
plz always double check everything, that it is same as that in console. Otherwise there will be bugs later.